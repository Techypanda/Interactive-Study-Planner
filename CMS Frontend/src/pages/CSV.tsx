import { Box, Button, CircularProgress, Container, Typography } from "@material-ui/core";
import { LocalShipping } from "@material-ui/icons";
import axios, { AxiosError } from "axios";
import { createRef, useState } from "react";
import { useQueryClient } from "react-query";
import styled from "styled-components"

function CSV(props: DefaultProps) {
  const csvFile = createRef<HTMLInputElement>()
  const client = useQueryClient()
  const [loading, setLoading] = useState(false);
  function handlePDFUpload(e: React.ChangeEvent<HTMLInputElement>) {
    setLoading(true);
    if (e.target.files) {
      const file = e.target.files[0]
      const reader = new FileReader()
      reader.readAsText(file)
      reader.onload = () => {
        if (reader.result !== "") {
          const csvData = btoa(unescape(encodeURIComponent(reader.result as string)))
          axios.post(`${process.env.REACT_APP_CSV_API}/add`, JSON.stringify({
            csvData: csvData
          }), {
            headers: {
              'Content-Type': 'application/json',
              'Authorization': `${client.getQueryData("token")}`
            }
          }).then((resp) => {
            alert(resp.data)
          }).catch((err) => {
            alert((err as AxiosError).response?.data)
          }).finally(() => {
            setLoading(false);
          })
        } else {
          alert("CSV File is empty")
          setLoading(false);
        }
      }
      reader.onerror = (err) => {
        alert(err)
      }
    }
  }

  return (
    <Box className={props.className} position="absolute" top={0} left={0} justifyContent="center" alignItems="center" minHeight="100vh" display="flex" minWidth="100vw" zIndex={-1}>
      <Container>
        {!loading ? <Box>
          <Typography align="center" variant="h4">Hi, Please Upload The CSV below and we will validate and perform the operation/s</Typography>
          <Box display="flex" justifyContent="center" mt={2}>
            <Button variant="contained" color="primary" onClick={() => csvFile?.current?.click()}>
              Upload <LocalShipping className="ml" />
            </Button>
            <input type="file" id="file" ref={csvFile} style={{ display: 'none' }} onChange={handlePDFUpload} />
          </Box>
        </Box> : <Box display="flex" justifyContent="center">
          <CircularProgress size={150} />
        </Box> }
      </Container>
    </Box>
  )
}
export default styled(CSV)`
.ml {
  margin-left: 10px;
}
`;