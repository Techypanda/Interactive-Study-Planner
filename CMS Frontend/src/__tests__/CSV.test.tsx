import React from 'react';
import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import CSV from '../pages/CSV';

describe("CSV Works when", () => {
  const client = new QueryClient();
  it("Renders Without A Crash", () => {
    render(
      <QueryClientProvider client={client}>
        <CSV />
      </QueryClientProvider>
    )
  })
  it("Meets Snapshot Specs", () => {
    const x = render(
      <QueryClientProvider client={client}>
        <CSV />
      </QueryClientProvider>
    )
    expect(x).toMatchSnapshot();
  })
})