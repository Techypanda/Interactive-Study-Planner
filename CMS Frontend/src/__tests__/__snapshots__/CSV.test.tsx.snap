// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV Works when Meets Snapshot Specs 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-4 sc-bdnxRM kkuJVC"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            class="MuiBox-root MuiBox-root-5"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
            >
              Hi, Please Upload The CSV below and we will validate and perform the operation/s
            </h4>
            <div
              class="MuiBox-root MuiBox-root-6"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Upload 
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root ml"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 8h-3V4H3c-1.1 0-2 .9-2 2v11h2c0 1.66 1.34 3 3 3s3-1.34 3-3h6c0 1.66 1.34 3 3 3s3-1.34 3-3h2v-5l-3-4zM6 18.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm13.5-9l1.96 2.5H17V9.5h2.5zm-1.5 9c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <input
                id="file"
                style="display: none;"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-4 sc-bdnxRM kkuJVC"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiBox-root MuiBox-root-5"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
          >
            Hi, Please Upload The CSV below and we will validate and perform the operation/s
          </h4>
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Upload 
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root ml"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 8h-3V4H3c-1.1 0-2 .9-2 2v11h2c0 1.66 1.34 3 3 3s3-1.34 3-3h6c0 1.66 1.34 3 3 3s3-1.34 3-3h2v-5l-3-4zM6 18.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm13.5-9l1.96 2.5H17V9.5h2.5zm-1.5 9c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <input
              id="file"
              style="display: none;"
              type="file"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
